Mobile Buy SDK/Mobile Buy SDK/Additions/NSDateFormatter+BUYAdditions.m

```diff
@@ -38,8 +38,29 @@ + (NSDateFormatter*)dateFormatterForShippingRates
 + (NSDateFormatter*)dateFormatterForPublications
 {
 	NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+	
+	if (![self is24hTimeFormat]) {
+		[dateFormatter setLocale:[NSLocale localeWithLocaleIdentifier:@"en_GB"]];
+	}
+	
 	dateFormatter.dateFormat = @"yyyy-MM-dd'T'HH:mm:ssZ";
 	return dateFormatter;
 }
 
++ (BOOL)is24hTimeFormat
+{
+	NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+	[formatter setLocale:[NSLocale currentLocale]];
+	[formatter setDateStyle:NSDateFormatterNoStyle];
+	[formatter setTimeStyle:NSDateFormatterShortStyle];
+	
+	NSString *dateString = [formatter stringFromDate:[NSDate date]];
+	
+	NSRange amRange = [dateString rangeOfString:[formatter AMSymbol]];
+	NSRange pmRange = [dateString rangeOfString:[formatter PMSymbol]];
+	BOOL is24h = (amRange.location == NSNotFound && pmRange.location == NSNotFound);
+	
+	return is24h;
```

Comment:
The issue seems to be that the `NSDateFormatter` cannot parse the date string because the format provided by the server doesn't match the one on the client. If that's the case, the solution would be for the server to provide consistent date formats. Do you have an example of the JSON response for the server that breaks using the current implementation?